Javascript javascript2 week1

hassou2n opened the pull request.

ncryer commented on javascript/javascript2/week1/hyfBya/main.js:
```diff
@@ -0,0 +1,25 @@
+function renderProducts(products) {
+  for (let i = 0; i < products.length; i++) {
+    let product = document.createElement("li");
+    product.innerHTML =
```
Great use of `innerHtml`! 
 
You might be interested in knowing that you can make life a little easier with string interpolation, like: 
```js
product.innerHTML = `
        <h3>Title: ${products[i].name}</h3>
        <p>Price: ${products[i].price}</p>
        <p>Rating: ${product[i].rating}</p>
        `;
```

ncryer commented on javascript/javascript2/week1/index.js:
```diff
@@ -0,0 +1,101 @@
+
+let danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+danishWords = danishWords.reduce((acc, curr) => {
+  return acc.length < curr.length ? acc : curr;
```
Beautiful use of `reduce`! Super clever, well done

ncryer commented on javascript/javascript2/week1/index.js:
```diff
@@ -0,0 +1,101 @@
+      letterÆ.length
+    }, ø:${letterØ.length}, å:${letterÅ.length} `
+  );
+}
```
Perhaps there's a way to do this dynamically? Instead of writing one function for every test, think about how we can write one function that solves this problem generally. 
 
Start with an array of things you want to search for: `const lettersToFind = ['æ', 'ø', 'å'];` 
Make an object to hold your results: `const results = {};` 
 
Now we just have to search through the letters of the input string, and for each letter we check if it is one of the letters we're looking for, e.g. `lettersToFind.includes(letter) // boolean` 
 
If it is, we check if the letter doesn't yet exist in our `results` object like so: `if (!results[letter]) { ... }` 
 
If it exists, we increment its count: `results[letter] = results[letter] + 1;` else we initialize it to 1: `results[letter] = 1;` 
 
Now we just have to count up the values in our results object and add that as a key to the object, then return the object. 
 
Write me on Slack if you have questions for how to proceed :)

ncryer commented on javascript/javascript2/week1/index.js:
```diff
@@ -0,0 +1,101 @@
+const getRandomNumber = (max) => Math.floor(Math.random() * max);
+const getRandomAname = () =>
+  `${animalNameG1[getRandomNumber(animalNameG1.length)]} ${
+    animalNameG2[getRandomNumber(animalNameG2.length)]
```
Excellent!!! Very good idea splitting this out into its own function! Well done!

ncryer commented on javascript/javascript2/week1/index.js:
```diff
@@ -0,0 +1,101 @@
+mybutton.onclick = function fname(event) {
+  const section = document.getElementById("section");
+  const p = document.createElement("p");
+  p.innerText = inputText + " " + "-" + " " + getRandomAname();
```
You can make this easier for yourself by writing ```p.innerText = `${inputText} - ${getRandomAname()}```

ncryer commented on javascript/javascript2/week1/index.js:
```diff
@@ -0,0 +1,101 @@
+  p.innerText = inputText + " " + "-" + " " + getRandomAname();
+
+  if (inputText.value != ""){
+    section.appendChild(p);
```
Very good! However if you'd like the section to not add a bunch of `<p>` tags each time the function is called, do 
```javascript 
section.innerHtml = `<p>${inputText} - ${getRandomAname()}</p>` 
```

In this way we simply overwrite whatever is in there, which means all new names appear on the same line :)

hassou2n commented:
Sorry I didn't do it this way, but I don't know why the right file wasn't
uploaded, I'm sorry. I'll do it

On Fri, 28 Jan 2022 at 15.20, Nicolai ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In javascript/javascript2/week1/index.js
> <https://github.com/hassou2n/hyf-homework/pull/15#discussion_r794457953>:
>
> > +}
>
> +notThisFunctionName(danishString); // returns {total: 1, å: 1,}
>
> +
>
> +const danishString2 = "Blå grød med røde bær";
>
> +
>
> +function notThisFunctionName2(params) {
>
> +  let letterÆ = params.split(" ").includes((e) => e === "æ");
>
> +  let letterØ = params.split(" ").filter((e) => e === "ø");
>
> +  let letterÅ = params.split(" ").filter((e) => e === "å");
>
> +
>
> +  return console.log(
>
> +    `total: ${letterÆ.length + letterØ.length + letterÅ.length} æ:${
>
> +      letterÆ.length
>
> +    }, ø:${letterØ.length}, å:${letterÅ.length} `
>
> +  );
>
> +}
>
>
> Perhaps there's a way to do this dynamically? Instead of writing one
> function for every test, think about how we can write one function that
> solves this problem generally.
>
> Start with an array of things you want to search for: const lettersToFind
> = ['æ', 'ø', 'å'];
> Make an object to hold your results: const results = {};
>
> Now we just have to search through the letters of the input string, and
> for each letter we check if it is one of the letters we're looking for,
> e.g. lettersToFind.includes(letter) // boolean
>
> If it is, we check if the letter doesn't yet exist in our results object
> like so: if (!results[letter]) { ... }
>
> If it exists, we increment its count: results[letter] = results[letter] +
> 1; else we initialize it to 1: results[letter] = 1;
>
> Now we just have to count up the values in our results object and add that
> as a key to the object, then return the object.
>
> Write me on Slack if you have questions for how to proceed :)
> ------------------------------
>
> In javascript/javascript2/week1/index.js
> <https://github.com/hassou2n/hyf-homework/pull/15#discussion_r794458479>:
>
> > +  "Rhiblotsa",
>
> +  "Doneoslead",
>
> +  "Octetsire",
>
> +  "Kreetsatee",
>
> +  "Pakoose",
>
> +  "Butsboon",
>
> +  "Roocseadine",
>
> +  "Xouggoo",
>
> +  "Taclaud",
>
> +  "Buffuin",
>
> +];
>
> +
>
> +const getRandomNumber = (max) => Math.floor(Math.random() * max);
>
> +const getRandomAname = () =>
>
> +  `${animalNameG1[getRandomNumber(animalNameG1.length)]} ${
>
> +    animalNameG2[getRandomNumber(animalNameG2.length)]
>
>
> Excellent!!! Very good idea splitting this out into its own function! Well
> done!
> ------------------------------
>
> In javascript/javascript2/week1/index.js
> <https://github.com/hassou2n/hyf-homework/pull/15#discussion_r794537693>:
>
> > +  }`;
>
> +
>
> +const setRandomName = () => {
>
> +  document.write.innerText = getRandomAname();
>
> +};
>
> +
>
> +document.querySelector("button").addEventListener("click",setRandomName);
>
> +setRandomName();
>
> +
>
> +const mybutton = document.getElementById("button");
>
> +let inputText;
>
> +
>
> +mybutton.onclick = function fname(event) {
>
> +  const section = document.getElementById("section");
>
> +  const p = document.createElement("p");
>
> +  p.innerText = inputText + " " + "-" + " " + getRandomAname();
>
>
> You can make this easier for yourself by writing p.innerText =
> `${inputText} - ${getRandomAname()}
> ------------------------------
>
> In javascript/javascript2/week1/index.js
> <https://github.com/hassou2n/hyf-homework/pull/15#discussion_r794542078>:
>
> > +  document.write.innerText = getRandomAname();
>
> +};
>
> +
>
> +document.querySelector("button").addEventListener("click",setRandomName);
>
> +setRandomName();
>
> +
>
> +const mybutton = document.getElementById("button");
>
> +let inputText;
>
> +
>
> +mybutton.onclick = function fname(event) {
>
> +  const section = document.getElementById("section");
>
> +  const p = document.createElement("p");
>
> +  p.innerText = inputText + " " + "-" + " " + getRandomAname();
>
> +
>
> +  if (inputText.value != ""){
>
> +    section.appendChild(p);
>
>
> Very good! However if you'd like the section to not add a bunch of <p>
> tags each time the function is called, do
>
> section.innerHtml = `<p>${inputText} - ${getRandomAname()}</p>`
>
> In this way we simply overwrite whatever is in there, which means all new
> names appear on the same line :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hassou2n/hyf-homework/pull/15#pullrequestreview-866075426>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVHH6LR3H6ML4YAYKD22LSLUYKQZBANCNFSM5M6LZBJA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

hassou2n commented:
I'll fix it, thank you for feedback

On Fri, 28 Jan 2022 at 12.39, Nicolai ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In javascript/javascript2/week1/hyfBya/main.js
> <https://github.com/hassou2n/hyf-homework/pull/15#discussion_r794431051>:
>
> > @@ -0,0 +1,25 @@
> +console.log("Script loaded");
> +
> +const products = getAvailableProducts();
> +console.log(products);
> +
> +const pUl = document.createElement("ul");
> +pUl.classList.add("products-list");
> +document.querySelector("main").appendChild(pUl);
> +
> +function renderProducts(products) {
> +  for (let i = 0; i < products.length; i++) {
> +    let product = document.createElement("li");
> +    product.innerHTML =
>
> Great use of innerHtml!
>
> You might be interested in knowing that you can make life a little easier
> with string interpolation, like:
>
> product.innerHTML = `        <h3>Title: ${products[i].name}</h3>        <p>Price: ${products[i].price}</p>        <p>Rating: ${product[i].rating}</p>        `;
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hassou2n/hyf-homework/pull/15#pullrequestreview-866036826>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVHH6LSZ5FKOKFBVO3FVZ3LUYJ6A7ANCNFSM5M6LZBJA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
